tasks/faucet.ts

```diff
@@ -1,12 +1,13 @@
-const fs = require("fs");
+import fs from "fs";
+import { task } from "hardhat/config";
 
 // This file is only here to make interacting with the Dapp easier,
 // feel free to ignore it if you don't need it.
 
 task("faucet", "Sends ETH and tokens to an address")
   .addPositionalParam("receiver", "The address that will receive them")
-  .setAction(async ({ receiver }, { ethers }) => {
-    if (network.name === "hardhat") {
+  .setAction(async ({ receiver }, { ethers, config }) => {
+    if (config.defaultNetwork === "hardhat") {
```

Comment:
I don't think this is correct, the default network will almost surely be hardhat all the time.

Reply:
You are also correct with that one. It should have been:
```js
task("faucet", "Sends ETH and tokens to an address")
  .addPositionalParam("receiver", "The address that will receive them")
  .setAction(async ({ receiver }, { ethers, network }) => {
    if (network.name === "hardhat") {
      // ...
    }
    // ...
```
Sorry for the inconvenience and thanks for your thorough review. I didn't know that there was a `network` object available and I also couldn't find any reference implementation which used it. Next time I will read the docs more obsessively!

Reply:
I've tested it :)